app/README.md

```diff
@@ -1,80 +1,118 @@
+Other scripts:
 
-## Package.json
+- `npm run storybook-build` - Exports a static site to `storybook-static/`
 
-### Scripts
+## 🐛 Testing
 
-These can be run with `npm run [name of script]`. For more about npm scripts, see [here](https://docs.npmjs.com/cli/v8/using-npm/scripts).
+[Jest](https://jestjs.io/docs/getting-started) is used as the test runner and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro) provides React testing utilities.
+
+Tests are manged as `.test.ts` (or `.tsx`) files in the the `tests/` directory.
+
+To run tests:
+
+- `npm test` - Runs all tests and outputs test coverage report
+- `npm run test-update` - Updates test snapshots
+- `npm run test-watch` - Runs tests in watch mode
+
+A subset of tests can be ran by passing a pattern to the script. For example, to only run tests in `tests/pages/`:
+
+```sh
+npm run test-watch -- pages
+```
 
-- `build`: Builds the app.
-- `dev`: Runs the app in development mode.
-- `format`: Instructs prettier to rewrite files with fixes for formatting violations.
-- `format-check`: Instructs prettier to only check files for violations without fixing them.
-- `lint`: Runs ESLint and reports warnings and errors.
-- `postinstall`: Copies the USWDS static assets into the project. `postinstall` runs automatically after `install`.
-- `start`: Runs the app in production mode.
-- `storybook`: Starts storybook.
-- `storybook-build`: Builds the static content for storybook.
-- `test`: Runs `jest --ci --coverage`. [--ci option](https://jestjs.io/docs/cli#--ci) is provided to prevent automatic creation of snapshots. This requires Jest to be run with `--updateSnapshot`. [--coverage option](https://jestjs.io/docs/cli#--coverageboolean) is provided to instruct jest to collect and report test coverage in output.
-- `ts:check`: Runs `tsc --noEmit`. [--noEmit option](https://www.typescriptlang.org/tsconfig#noEmit) is provided to prevent type checker compiler from outputting files.
+## 🤖 Type checking, linting, and formatting
+
+- [TypeScript](https://www.typescriptlang.org/) is used for type checking.
+  - `npm run ts:check` - Type checks all files
+- [ESLint](https://eslint.org/) is used for linting. This helps catch common mistakes and encourage best practices.
+  - `npm run lint` - Lints all files and reports any errors
+  - `npm run lint-fix` - Lints all files and fixes any auto-fixable errors
+- [Prettier](https://prettier.io/) is used for code formatting. This reduces the need for manual formatting or nitpicking and enforces a consistent style.
+  - `npm run format`: Formats all files
+  - `npm run format-check`: Check files for formatting violations without fixing them.
+
+It's recommended that developers configure their code editor to auto run these tools on file save. Most code editors have plugins for these tools or provide native support.
```

Comment:
This may be obvious but in the vscode case developers would also have to add the extensions for prettier, eslint, and typescript right? 

should we note that somewhere?

Reply:
Yea true, I'll add that to the `details` part

Reply:
Added a note in https://github.com/navapbc/template-application-nextjs/pull/67/commits/48b62ab7c8f78149bce1a3cca951e33500af8b36